### PR TITLE
ENH: TractOracle module

### PR DESCRIPTION
**Pull request by levje**
## Describe your changes

This is completely unrelated to the module I was supposed to do (for which I should create a PR very soon), but I took some time to write a simple tracking module using RL-based tractography. It was easier for me as a practice, especially for a proper nf-core and nf-test setup, to write a module on a script that I knew more about, which links to my most recent MedIA paper.

We were discussing what it would take for us to use the TractOracle framework in our pipelines and how to make it accessible to other collaborators or members of the lab. I thought making a module would be a quick way to access and use such a framework. Please, let me know if this doesn't fit the scope of nf-neuro.

Note: This module works fine on CPU, but it **should** take advantage of GPU (via the `--gpus all` or `--nv`), however I am not sure what would be the best way to indicate this on nf-neuro.

## List test packages used by your module
Mostly scilpy and TractOracle-IRT.

## Checklist before requesting a review

- Create the tool:
  - [ x ] Edit `./modules/nf-neuro/<category>/<tool>/main.nf`
  - [ x ] Edit `./modules/nf-neuro/<category>/<tool>/meta.yml`
  - [ x ] Edit `./modules/nf-neuro/<category>/<tool>/environment.yml`
- Generate the tests:
  - [ x ] Edit `./modules/nf-neuro/<category>/<tool>/tests/main.nf.test`
  - [ x ] Run the tests to generate the `main.nf.test.snap` snapshots
- Ensure the syntax is correct :
  - [ x ] Run `prettier` and `editorconfig-checker` to fix common syntax issues
  - [ ] Run `nf-core modules lint` and fix all errors
  - [ x ] Ensure your variables have good, clear names

### CODE REVIEW

File: modules/nf-neuro/tracking/tractoracle/main.nf [hunk 1]
```diff
@@ -0,0 +1,66 @@
+process TRACKING_TRACTORACLE {
+    tag "${meta.id}"
+    label 'process_single'
+
+    container 'mrzarfir/tractoracle-irt:base'
```
[the remark this answers is not in the document]
Comment (levje) [reply modules/nf-neuro/tracking/tractoracle/main.nf:5]: Everything should be running with GPU (strongly recommended!!), otherwise it's going to take forever on real subjects. Since it's using PyTorch in any case, not sure if it's worth making two images, but I might be wrong!

File: modules/nf-neuro/tracking/tractoracle/main.nf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def compress = task.ext.compress ? "${task.ext.compress}" : "0.1"
+    def n_actor = task.ext.n_actor ? "${task.ext.n_actor}" : "10000"
+    def npv = task.ext.npv ? "${task.ext.npv}" : "1"
+    def agent_checkpoint = task.ext.agent_checkpoint ? "${task.ext.agent_checkpoint}" : "public://sac_irt_inferno"
+    def min_length = task.ext.min_length ? "${task.ext.min_length}" : "20"
+    def max_length = task.ext.max_length ? "${task.ext.max_length}" : "200"
```
Comment (AlexVCaron) [inline modules/nf-neuro/tracking/tractoracle/main.nf:25]: Add those `args` to the **meta.yml** file. Sorry, missed it 😅

File: modules/nf-neuro/tracking/tractoracle/main.nf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+
+    touch ${prefix}__interface.nii.gz
+    touch ${prefix}__tracking.trk
+
```
Comment (AlexVCaron) [inline modules/nf-neuro/tracking/tractoracle/main.nf:59]:
```suggestion
uv run scil_tracking_pft_maps.py -h
```

Also add a call to the **usage** of `tractoracle_irt/runners/ttl_track.py` in there.

File: modules/nf-neuro/tracking/tractoracle/meta.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+name: tracking_tractoracle
+description: Compute the interface and generate a tractogram using TractOracle-IRT,
+  a deep RL-based tractography algorithm.
```
Comment (AlexVCaron) [inline modules/nf-neuro/tracking/tractoracle/meta.yml:3]: Do you know of any specific use-cases that would require tailored configuration of the module ? If so, add them to the description for users knowledge. If you tailored some configurations, it's also where you present and describe them. You can write the description on multiple line by adding a pipe `|` to the section, aka `description: |` and writing on the line(s) below. The documentation will render it to its best, it's usually fine.

Another good way to do so is create **test cases per use-case**, with an associated `nextflow_<use-case-name>.config` in the module's **tests** directory. Then you can then refer to those configuration files in the description (a simple highlight, or a href to the file(s) that will be added to nf-neuro once the PR is merged), instead of listing parameters explicitly.

File: modules/nf-neuro/tracking/tractoracle/meta.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      description: "TractOracle-IRT: Exploring the robustness of TractOracle methods in RL-based tractography."
+      homepage: "https://github.com/scil-vital/tractoracle_irt"
+      doi: "10.1016/j.media.2025.103743"
+      identifier: ""
+  - scilpy:
+      description: "The Sherbrooke Connectivity Imaging Lab (SCIL) Python dMRI processing
+        toolbox."
+      homepage: "https://github.com/scilus/scilpy.git"
+      identifier: ""
+
```
Comment (AlexVCaron) [inline modules/nf-neuro/tracking/tractoracle/meta.yml:19]: See above comment to add `args`

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Second round of comment looking at metadatas. Once addressed, it will be GTG 🚢

**Comment by levje** [comment]
On it, I somehow missed your second round of comments.

File: modules/nf-neuro/tracking/tractoracle/meta.yml [hunk 6]
```diff
@@ -0,0 +1,130 @@
+name: tracking_tractoracle
+description: |
+  Compute the interface and generate a tractogram using TractOracle-IRT,
+  a deep RL-based tractography algorithm. As this is a RL-based method,
+  we provide through this module the best(s) pre-trained agent(s) available
+  for tractography.
```
Comment (AlexVCaron) [inline modules/nf-neuro/tracking/tractoracle/meta.yml]: Some writing suggestions from my outsider POV, feel free to take what you want !


```suggestion
  Compute the interface between white matter (WM) and grey matter (GM) for seeding and
  generate a tractogram using TractOracle-IRT, a deep Reinforcement Learning (RL) based
  tractography algorithm. Through this module, we provide the best(s) pre-trained agent(s)
  available for tractography using this RL-based method.
```
